src/app/config/constants/constants.ts

```diff
@@ -1,26 +1,35 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 
 import apiURL from 'src/app/config/constants/apiURL';
+
 interface externalNameResponseFormat {
-  externalName: String;
+  externalName: string;
 };
 
-@Injectable({
-  providedIn: 'root',
-})
+@Injectable({ providedIn: 'root' })
 export class DoubtfireConstants {
-  constructor(private httpClient: HttpClient) {
-    httpClient.get(`${this.apiURL}/settings`).toPromise().then((response: externalNameResponseFormat) => {
-      this.externalName = response.externalName;
-    })
+  constructor(private http: HttpClient) {
```

Comment:
Should we load in constructor of OnInit?

Reply:
Injectibles (services) don't have use the OnInit lifecycle, you use the constructor for that -- any other component should use the ngOnInit